Interleaved removal

yuzefovich opened the pull request:
Depends on #70239.

yuzefovich committed:
colbuilder: remove the check for index join on the interleaved index

Interleaved indexes are no more.

Release note: None

yuzefovich committed:
sql: remove the interleaved logic from fetchers

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70235)
<!-- Reviewable:end -->

yuzefovich committed:
row: simplify fetcher to only handle a single table

Previously, `row.Fetcher` could handle multiple tables at the same time.
This was needed to support the interleaved tables efficiently, but since
the interleaved tables are no more, we can simplify the fetcher now.

There are more things that can be removed from the fetcher about
handling of the interleaved tables, but I chose to defer them to a later
commit. This commit's main focus is simplifying the fetcher to work
under the assumption that it will be fetching only from a single table.

Release note: None
